### PR TITLE
fix(river): enable/disable river activity on user ban/(un)ban

### LINKED ISSUE #12514: Elgg 3: Hide river activity for disabled plugin
**Let's say I have the blog plugin enabled and I created a blog post and the river activity looks like this.**
image

-----------

**Now I disabled the blog plugin and the river activity is shown like this.**
image

### PR DESCRIPTION
**Pull request by rohit1290**
Fix #12514

### CODE REVIEW

File: engine/classes/ElggUser.php [hunk 1]
```diff
@@ -127,6 +127,8 @@ public function ban($reason = '') {
 		if (!_elgg_services()->events->trigger('ban', 'user', $this)) {
 			return false;
 		}
+		
+		_elgg_services()->events->trigger('disable:after', 'all', $this);
```
Comment (jeabakker) [inline engine/classes/ElggUser.php]: this is a completely wrong event at this time. nothing is getting disabled.
The correct way would be to trigger an after event after the user is actually banned in the DB

File: engine/classes/ElggUser.php [hunk 2]
```diff
@@ -127,6 +127,8 @@ public function ban($reason = '') {
...
 
 		$this->ban_reason = $reason;
 		$this->banned = 'yes';
```
Comment (jeabakker) [inline engine/classes/ElggUser.php]: maybe here
`_elgg_services()->events->triggerAfter('ban', 'user', $this)`

File: engine/classes/ElggUser.php [hunk 3]
```diff
@@ -150,6 +152,8 @@ public function unban() {
 		if (!_elgg_services()->events->trigger('unban', 'user', $this)) {
 			return false;
 		}
+		
+		_elgg_services()->events->trigger('enable:after', 'all', $this);
```
Comment (jeabakker) [inline engine/classes/ElggUser.php]: wrong event, see comment above

**Review by jeabakker — CHANGES_REQUESTED** [review]
Then with the new events maybe enable/disable river activity from the users

File: engine/classes/ElggUser.php (hunk 3 above)
Comment (rohit1290) [reply engine/classes/ElggUser.php]: please check the updated commit

File: engine/classes/ElggUser.php (hunk 2 above)
Comment (rohit1290) [reply engine/classes/ElggUser.php]: please check the updated commit

File: engine/classes/ElggUser.php (hunk 1 above)
Comment (rohit1290) [reply engine/classes/ElggUser.php]: please check the updated commit

**Comment by jdalsem** [comment]
This will not be merged. See reason https://github.com/Elgg/Elgg/issues/12514#issuecomment-482527668

**Pull request closed without merging**